stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/ChatInitializer.kt

```diff
@@ -5,33 +5,40 @@ import com.google.firebase.FirebaseApp
 import io.getstream.chat.android.client.ChatClient
 import io.getstream.chat.android.client.logger.ChatLogLevel
 import io.getstream.chat.android.client.notifications.handler.NotificationConfig
+import io.getstream.chat.android.client.notifications.handler.NotificationHandlerFactory
 import io.getstream.chat.android.core.internal.InternalStreamChatApi
 import io.getstream.chat.android.livedata.ChatDomain
 import io.getstream.chat.android.pushprovider.firebase.FirebasePushDeviceGenerator
-import io.getstream.chat.android.pushprovider.huawei.HuaweiPushDeviceGenerator
 import io.getstream.chat.ui.sample.BuildConfig
 import io.getstream.chat.ui.sample.R
+import io.getstream.chat.ui.sample.feature.HostActivity
+
+
 
 @OptIn(InternalStreamChatApi::class)
 class ChatInitializer(private val context: Context) {
 
     @Suppress("UNUSED_VARIABLE")
     fun init(apiKey: String) {
         FirebaseApp.initializeApp(context)
+        NotificationHandlerFactory.newMessageIntent = {
```

Comment:
🤔 I think this might be a little bit confusing. We'll have two entry points for notifications:
1. You need to override `NotificationHandlerFactory.newMessageIntent` if you are using the default implementation
2. But you need to pass custom `NotificationHandler` to `ChatClient` if you want to customize the behavior
Maybe we could pass `NotificationHandlerFactory` instead of `NotificationHandler` to `ChatClient` or allow overriding `createNotificationHandler`, so they always interact with the factory? What do you think?

Reply:
Sounds like passing factory to LLC is a more direct way of customization in one place. Customers are free to override default implementation of factory if they want